### DISCUSSION TITLE
Xiaomi XyperOS XDrip lockscreen widget issues

### DISCUSSION THREAD

**Discussion by karzyg555** [NightscoutFoundation/xDrip#3564] [Bug reports]
Hi,

I use Xiaomi12 with XyperOS 1.0.9, but I believe this problem also appeared in the latest version of MIUI as well. I've created XDrip widget and it works fine on unlocked screen, but it is not displayed on locked screen. I've verified all permissions of XDrip app and I believe i've configured all settings to display widget on locked screen both in Android and in XDrip app (public notification and high priority notifications enabled). 
At the same time Always On XDrip notification works fine and Juggluco widget on locked screen also works fine. 
My version of XDrip is: 645e8b4-2024.04.19

**Comment by Navid200** [comment]
Please, first post a screenshot of the classic status page.

**Reply by karzyg555** [reply]
![xdrip-xioami](https://github.com/NightscoutFoundation/xDrip/assets/173735078/e64464ec-4b7f-445c-9d43-2474881ca0ca)
attached

**Reply by Navid200** [reply]
Thanks  

Have you followed all the steps here?  
https://navid200.github.io/xDrip/docs/Display/AOD.html

**Comment by karzyg555** [comment]
yes AOD feature works fine, the lock screen widget doesn't work for me.

**Reply by Navid200** [reply]
Please explain the distinction.

**Reply by karzyg555** [reply]
I'm talking about showing XDrip notification content (with graph) on lock screen the same way Juggluco does (attached). 
![notification](https://github.com/NightscoutFoundation/xDrip/assets/173735078/fd9868c4-55a9-4205-aa80-7ca8632083b1)

**Reply by Navid200** [reply]
> yes AOD feature works fine.  

That's perfect.  
If it works, can you use another phone and take a picture to show us what it looks like.  
  
If you look at this page, you can see that in a picture I have included:  https://navid200.github.io/xDrip/docs/Display/AOD.html  
In the very first picture on that page, you can see what xDrip does when AOD is enabled.  
Can you show what that looks like on your screen to make sure we understand each other?  A picture is worth a thousand words.

**Comment by karzyg555** [comment]
So on lock screen notification area shows only Juggluco notification channel
![xdrip2](https://github.com/NightscoutFoundation/xDrip/assets/173735078/1f2bd2a9-5b1d-44f7-a4f7-0f22698fc6c2)
and when i unlock i see both Juggluco and Xdrip
![xdrip1](https://github.com/NightscoutFoundation/xDrip/assets/173735078/ed168682-b455-48a4-8b90-64bfe731c35c)

Previously i've seen both on locked screen.

**Reply by Navid200** [reply]
I don't see the point if AOD works.  That's why I asked you to show AOD.  You aren't showing that.
But, you said AOD worked for you.

If you can see your glucose and trend arrow and curve on screen when your phone is sitting face up on the table all the time, why would you need to pick it up and press the power button to see the same thing?  
  
This is what xDrip AOD looks like: 
![AOD_Sample](https://github.com/NightscoutFoundation/xDrip/assets/51497406/a0d18144-3202-4b6f-aaa9-a015762bbc3c)

**Comment by karzyg555** [comment]
I just discover AOD feature today so it is enough for me :) I just wanted to report that notification channel on locked screen has worked previously for me and it stopped and in XDrip settings it looks it should be supported as there are configuration item for this feature,

**Reply by Navid200** [reply]
Thanks you for that.  
  
We are all volunteers.
There are so many open issues.  We have no choice but to prioritize and focus on items that are catastrophic before moving on to reasonable preferences/requests followed by good to have requests.  
  
That's why I was trying to figure out why this was needed.  
Thanks for reporting it.

**Comment by karzyg555** [comment]
Sure i completely understand, with AOD feature i'm done and very happy !

**Comment by zTeveg** [comment]
Try activating "Notification channel"
https://xdrip.readthedocs.io/en/latest/use/alarms/?h=channel#notification-channel
This was the solution for me after the "HyperOS" update 

And for me I really like to have both AOD and the lockscreen notification. Because when I pick up my phone, it jumps straight to the lockscreen.

**Reply by karzyg555** [reply]
Thanks, this is exactly what i wanted ! Much appreciated :)
